src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/services/PopulateIndexService.kt

```diff
@@ -92,7 +94,15 @@ class PopulateIndexService(
   fun populateIndexWithPrisoner(prisonerNumber: String): Either<Error, Prisoner> =
     indexStatusService.getIndexStatus()
       .failIf(IndexStatus::isNotBuilding) { BuildNotInProgressError(it) }
-      .flatMap { prisonerSynchroniserService.index(prisonerNumber, it.currentIndex.otherIndex()) }
+      .flatMap {
+        nomisService.getOffender(prisonerNumber)?.let { ob ->
+          prisonerSynchroniserService.index(ob, it.currentIndex.otherIndex())
+        }?.right() ?: run {
+          // can happen if a prisoner is deleted or merged once indexing has started
+          telemetryClient.trackPrisonerEvent(BUILD_PRISONER_NOT_FOUND, prisonerNumber)
```

Comment:
we weren't previously creating a telemetry event for this, but the old prisoner search did so have duplicated that functionality here